src/components/MyAvatarMenu/RecentTransactionsModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { formatRelative } from 'date-fns';
+import { FiExternalLink } from 'react-icons/fi';
+
+import { Box, Flex, Link, Modal, Text } from 'ui';
+
+const TX_LIST_KEY = 'capeRecentTxs';
+
+type Tx = {
+  timestamp: number; // Date.now() format
+  description: string;
+  hash?: string;
+  status: 'pending' | 'confirmed' | 'error';
+  chainId: string;
+};
+
+const getTxList = (): Tx[] =>
+  JSON.parse(localStorage.getItem(TX_LIST_KEY) || '[]').slice(0, 10);
+
+const saveTxList = (list: Tx[]) =>
+  localStorage.setItem(TX_LIST_KEY, JSON.stringify(list));
+
+export const addTransaction = (tx: Tx) => saveTxList([tx, ...getTxList()]);
+
+export const updateTransaction = (timestamp: number, updates: Partial<Tx>) => {
+  const list = getTxList();
+  const index = list.findIndex(tx => tx.timestamp === timestamp);
+  if (index === -1) return;
+  list.splice(index, 1, { ...list[index], ...updates });
```

Comment:
rather than messing with the index, it might be cleaner to just use `Array.prototype.find`, and allow the object reference to do the heavy lifting. Just an idea